app/main.py

```diff
@@ -2,5 +2,19 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    results_storage = {}
+
+    def inner_checker(*args) -> str:
+
+        key = args
+        if key in results_storage:
+            print("Getting from cache")
+            return results_storage[key]
+        else:
+            print("Calculating new result")
+            value = func(*args)
+            results_storage[key] = value
+            return value
```

Comment:
No need to create additional `value` variable here. After this fix you could also move return outsite of if/else to avoid return duplicating